consensus/src/block_storage/block_store.rs

```diff
@@ -294,37 +258,29 @@ impl BlockStore {
 
         let block_tree = self.inner.clone();
         let storage = self.storage.clone();
-        let commit_root = self.commit_root();
 
-        self.inner
-            .write()
-            .update_ordered_root_id(block_to_commit.id());
-        update_counters_for_ordered_blocks(&blocks_to_commit);
-
-        // asynchronously execute and commit
+        // This callback is invoked synchronously withe coupled-execution and asynchronously in decoupled setup.
         self.state_computer
             .commit(
                 &blocks_to_commit,
                 finality_proof,
                 Box::new(
-                    move |executed_blocks: &[Arc<ExecutedBlock>],
+                    move |committed_blocks: &[Arc<ExecutedBlock>],
```

Comment:
We added some comments at the BufferManager that the callback should be valid if we merge the batches and only call the last callback. Maybe let's also mention it here.